options/posix/generic/posix_string.cpp

```diff
@@ -1,3 +1,5 @@
+#define _GNU_SOURCE
```

Comment:
Wrapping this in an `#ifndef _GNU_SOURCE` might solve the problem of compilers predefining this macro.

Reply:
Good idea, will do